### PR TITLE
Add setup files 

### PR DESCRIPTION
**Pull request by lucyleeow**
Install will not work (via `pip install`) on a clean/empty environment due to difficulties installing iris from pip (see https://github.com/SciTools/iris/issues/3777). Ultimate goal is to put package on conda-forge

cc @tjtg @dmentipl 

Testing:
 - [N/A] Ran tests and they passed OK
 - [N/A] Added new tests for the new feature(s)

CLA
 - [N/A] If a new developer, signed up to CLA

### CODE REVIEW

File: improver/__init__.py
Comment (dmentipl) [inline improver/__init__.py:34]: An alternative to defining the version here is to do it in the setup.cfg file. Then here you can do the following:

```python
import importlib_metadata
__version__ = importlib_metadata.version('improver')
```

File: setup.cfg
Comment (dmentipl) [inline setup.cfg:3]: See the comment above (in `__init__.py`).

**Review by dmentipl — COMMENTED** [review]
These changes look good to me. I've made some comments regarding setting the version. To me it makes sense to put the version number in `setup.cfg` rather than `__init__.py`. However, I think either way works similarly.

I suppose because the tests aren't passing yet it shouldn't be approved.

One other thing to think about is how the package versions listed in `setup.cfg` interact with the conda environment files in the "envs" directory. I suggest that the versions listed in `setup.cfg` should be minimum requirements, as you've done.

File: improver/__init__.py
Comment (lucyleeow) [reply improver/__init__.py:34]: I have no strong opinion either day, WDYT @tjtg ?

Comment (tjtg) [reply improver/__init__.py:34]: I don't have strong opinions about this, so happy to leave it as is.

**Review by tjtg — APPROVED** [review]
I'm happy with the changes here. Tests are now passing after c1e3d4c, so OK to merge when you're ready.

**Review by dmentipl — APPROVED** [review]

**Review by tjtg — APPROVED** [review]

**Review by dmentipl — APPROVED** [review]

**Pull request merged**
